pkg/dra/claims.go

```diff
@@ -0,0 +1,187 @@
+/*
+Copyright The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package dra
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"strconv"
+
+	corev1 "k8s.io/api/core/v1"
+	resourcev1beta1 "k8s.io/api/resource/v1beta1"
```

Comment:
Should this be v1beta2?

cc @johnbelamaric?

Reply:
What was our plan around handling graduation among the different k8s versions?

In 1.34, DRA will be GA so I think the v1 api could be used. I guess that 1.31 and 1.32 may need to use v1beta1? And 1.33 should use v1beta2?

And eventually we would want to switch to using v1 api for DRA but right now we have a bit of complicated support here.

Reply:
IMO supporting v1beta2 and v1 is enough. 1.33+ only. Simplifies the story. WDYT?

Reply:
cc @mimowo @tenzen-y